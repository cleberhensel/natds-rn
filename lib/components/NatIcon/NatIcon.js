import React from 'react';
import PropTypes from 'prop-types';
import { load as loadSvg } from '../../assets/icons/svg';
import { ThemeNaturaLight } from '../../../lib';

export const NatIcon = ({ name, size, fill }) => {
  const IconComponent = loadSvg(name);
  return (
    <IconComponent width={size} height={size} style={{ fill }} />
  );
};

NatIcon.propTypes = {
  name: PropTypes.string,
  fill: PropTypes.string,
  size: PropTypes.number,
};

NatIcon.defaultProps = {
  name: 'filled-brand-naturarosacea',
  fill: ThemeNaturaLight.palette.text.primary,
  size: 24,
};
