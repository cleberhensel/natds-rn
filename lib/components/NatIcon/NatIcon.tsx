import React from 'react';
import { load as loadSvg } from '../../assets/icons/svg';
import { ThemeNaturaLight } from '../../themes/ThemeNaturaLight';

type NatIconProps = {
  fill?: string,
  name?: string,
  size?: number
};
export const NatIcon: React.SFC<NatIconProps> = ({ name, size, fill }) => {
  const IconComponent = loadSvg(name);

  return <IconComponent width={size} height={size} style={{ fill }} />;
};
NatIcon.defaultProps = {
  fill: ThemeNaturaLight.palette.text.primary,
  name: 'filled-brand-naturarosacea',
  size: 24,
};
