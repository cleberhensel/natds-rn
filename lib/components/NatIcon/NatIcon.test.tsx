import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components/native";
import { NatIcon } from "./NatIcon";
describe("NatIcon component default", () => {
  it("Should render NatIcon component ", () => {
    const button = renderer.create(<NatIcon />).toJSON();
    expect(button).toMatchSnapshot("NatIcon snapshot");
  });
});
