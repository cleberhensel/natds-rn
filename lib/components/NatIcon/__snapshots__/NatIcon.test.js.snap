// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatIcon component default Should render NatIcon component  1`] = `
<SvgMock
  height={24}
  style={
    Object {
      "fill": "#333333",
    }
  }
  width={24}
/>
`;
